app/v1/controllers/validators/CreateAndAmendOtherDeductionsValidatorFactory.scala

```diff
@@ -0,0 +1,157 @@
+          parsed: CreateAndAmendOtherDeductionsRequestData): Validated[Seq[MtdError], CreateAndAmendOtherDeductionsRequestData] = {
+        import parsed.body._
+        List(
+          validateBodyFieldFormat(seafarers),
+          validateBodyDateRange(seafarers)
+        ).traverse(identity)
+          .map(_ => parsed)
+      }
+
+    }
+
+  private def validateBodyFieldFormat(seafarers: Option[Seq[Seafarers]]): Validated[Seq[MtdError], Unit] = {
+    val validationResult = seafarers
+      .getOrElse(Seq.empty[Seafarers])
+      .zipWithIndex
+      .traverse_ { case (item, index) => validateSeafarers(item, index) }
+
+    validationResult
+  }
+
+  private def validateBodyDateRange(seafarers: Option[Seq[Seafarers]]): Validated[Seq[MtdError], Unit] = {
+    val validationResult = seafarers
+      .getOrElse(Seq.empty[Seafarers])
+      .zipWithIndex
+      .traverse_ { case (item, index) => validateToDateBeforeFromDate(item, index) }
+
+    validationResult
+  }
+
+  private def validateSeafarers(seafarers: Seafarers, arrayIndex: Int): Validated[Seq[MtdError], Unit] = {
+    (
+      ValidateCustomerReference.validateOptional(field = seafarers.customerReference, path = s"/seafarers/$arrayIndex/customerReference"),
+      validateAmountDeducted(value = seafarers.amountDeducted, path = s"/seafarers/$arrayIndex/amountDeducted"),
+      validateNameOfShip(
+        field = seafarers.nameOfShip,
+        path = s"/seafarers/$arrayIndex/nameOfShip"
+      ),
+      ValidateDate.validateDateFormat(
+        field = seafarers.fromDate,
+        path = s"/seafarers/$arrayIndex/fromDate"
+      ),
+      ValidateDate.validateDateFormat(
+        field = seafarers.toDate,
+        path = s"/seafarers/$arrayIndex/toDate"
+      )
+    ).tupled
+      .andThen { case (_, _, _, _, _) => Validated.Valid(()) }
+  }
+
+  object ValidateCustomerReference {
+    val customerRefRegex: String = "^[0-9a-zA-Z{À-˿’}\\- _&`():.'^]{1,90}$"
+
+    def validateOptional(field: Option[String], path: String): Validated[Seq[MtdError], Unit] = {
+
+      field match {
+        case None        => Valid(())
+        case Some(value) => validate(value, path)
+      }
+    }
+
+    private def validate(customerRef: String, path: String): Validated[Seq[MtdError], Unit] = {
+      if (customerRef.matches(customerRefRegex)) Valid(()) else Invalid(List(CustomerReferenceFormatError.copy(paths = Some(Seq(path)))))
+    }
+
+  }
+
+  private val resolveAmountDeducted = ResolveParsedNumber()
+
+  private def validateAmountDeducted(value: BigDecimal, path: String): Validated[Seq[MtdError], Unit] = {
+    resolveAmountDeducted(value, path = Some(path)).map(_ => ())
+  }
+
+  private def validateNameOfShip(field: String, path: String): Validated[Seq[MtdError], Unit] = {
+    if (field.length <= 105) Valid(()) else Invalid(List(NameOfShipFormatError.copy(paths = Some(Seq(path)))))
+  }
+
+  object ValidateDate {
+    val dateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd")
```

Comment:
can be private. But also, can just import java.time.format.DateTimeFormatter.ISO_LOCAL_DATE instead.